source/language/openpulse.rst

```diff
@@ -290,28 +313,261 @@ pushing the results into some buffer which is then accessed outside the program.
+  }
+
+  defcal geo_gate(angle[32]: theta) $0 {
+      // theta: rotation angle (about z-axis) on Bloch sphere
+
+      tx_channel dq = txch($q, “drive”);
+
+      // Assume we have calibrated 0->1 pi pulses and 1->2 pi pulse
+      // envelopes (no sideband)
+      waveform X_01 = {...};
+      waveform X_12 = {...};
+      float[32] a = sin(theta/2);
+      float[32] b = sqrt(1-a**2);
+
+      // Double-tap
+      play(dq, scale(a, X_01), frame_01);
+      play(dq, scale(b, X_12), frame_12);
+      play(dq, scale(a, X_01), frame_01);
+      play(dq, scale(b, X_12), frame_12);
+  }
+
+Neutral atoms
+~~~~~~~~~~~~~
+
+In this simple example, the signal chain is composed of two electro-optic modulators (EOM) and
+an acousto-optic deflector (AOD). The EOMs put sidebands on the laser light while the AOD diffracts
+the light in an amount proportional to the frequency of the RF drive. This example was chosen
+because it is similar in spirit to the work by Levine et al.:cite:`levine2019` except that phase
+control is exerted using virtual Z gates on the AODs -- requiring frame tracking of the qubit
+frequency yet application of a tone that maps to the qubit position (i.e. requires the use of a
+sideband).
+
+The program aims to perform a Hahn echo sequence on q1, and a Ramsey sequence on q2 and q3.
+
+.. code-block:: javascript
+
+  defcal neutral_atoms {
+    // Define the channels
+    eom_a_channel = txch(0, "eom_a");
+    eom_a_channel = txch(1, "eom_b");
+    aod_channel = txch(0, "aod");
+
+    // Define the Raman frames, which are detuned by an amount Δ from the  5S1/2 to 5P1/2 transition
+    // and offset from each other by the qubit_freq
+    raman_a_frame = Frame(Δ, 0.0)
+    raman_b_frame = Frame(Δ-qubit_freq, 0.0)
+
+    // Three copies of qubit freq to track phase of each qubit
+    q1_frame = Frame(qubit_freq, 0)
+    q2_frame = Frame(qubit_freq, 0)
+    q3_frame = Frame(qubit_freq, 0)
+
+    // Generic gaussian envelope
+    waveform π_half_sig = gaussian(..., π_half_time, ...)
+
+    // Waveforms ultimately supplied to the AODs. We mix our general Gaussian pulse with a sine wave to
+    // put a sideband on the signal construction to target the qubit position while maintainig the
+    // desired Rabi rate.
+    q1_π_half_sig = mix(π_half_sig, sine(q1_π_half_amp, q1_pos_freq-qubit_freq, 0.0, π_half_time));
+    q2_π_half_sig = mix(π_half_sig, sine(q2_π_half_amp, q2_pos_freq-qubit_freq, 0.0, π_half_time));
+    q3_π_half_sig = mix(π_half_sig, sine(q3_π_half_amp, q3_pos_freq-qubit_freq, 0.0, π_half_time));
+
+    for τ in [0: T]:
+        // Simultaneous π/2 pulses
+        play(eom_a_channel, constant(raman_a_amp, π_half_time) , raman_a_frame);
+        play(eom_b_channel, constant(raman_b_amp, π_half_time) , raman_b_frame);
+        play(aod_channel, q1_π_half_sig, q1_frame);
+        play(aod_channel, q1_π_half_sig, q2_frame);
+        play(aod_channel, q1_π_half_sig, q3_frame);
+
+        // Time delay all
+        delay(.*, τ/2)
```

Comment:
Rather than introducing new syntax, does it make sense for a no argument delay/barrier to delay all the frames in scope? That's how the spec is currently written for qubits.